### PR TITLE
Add CI job to add coverage to GH Action summary

### PR DESCRIPTION
**Pull request by zerolab**
So as to make the developer experience nicer.

e.g. https://github.com/ONSdigital/dis-wagtail/actions/runs/12260657615/attempts/1#summary-34205964707

<details><summary>🖼️ Screenshot</summary>

Screenshot 2024-12-10 at 16 45 28

</details>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -130,6 +130,48 @@ jobs:
...
         run: make test
+      - name: Upload coverage data
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-data
+          path: .coverage
+          if-no-files-found: ignore
+          include-hidden-files: true
+
+  coverage:
```
Comment (RealOrangeOne) [inline .github/workflows/ci.yml:141]: Question: Why is this a separate step rather than being handled in the `test-python` step?e

Comment (zerolab) [reply .github/workflows/ci.yml:141]: A personal preference and habit from package work

**Review by RealOrangeOne — APPROVED** [review]

**Review by AdamHawtin — APPROVED** [review]
Looks good 👍

**Review by sanjeevz3009 — COMMENTED** [review]
Reviewed!

**Pull request merged**